src/middlewared/middlewared/plugins/ups.py

```diff
@@ -178,11 +178,32 @@ def driver_choices(self):
         Str('notify_type')
     )
     async def upssched_event(self, notify_type):
+        config = await self.config()
+
         if notify_type.lower() == 'shutdown':
-            syslog.syslog(syslog.LOG_INFO, 'upssched-cmd "issuing shutdown"')
-            await run('/usr/local/sbin/upsmon', '-c', 'fsd', check=False)
+            # Before we start FSD with upsmon, lets ensure that ups is not ONLINE (OL).
+            # There are cases where battery/charger issues can result in ups.status being "OL LB" at the
+            # same time. This will ensure that we don't initiate a shutdown if ups is OL.
+            stats_output = (
+                await run(
+                    '/usr/local/bin/upsc', f'{config["identifier"]}@localhost:{config["remoteport"]}',
+                    check=False
+                )
+            ).stdout
+
+            ups_status = re.findall(
+                fr'ups.status: (.*)',
+                '' if not stats_output else stats_output.decode()
+            )
+            if ups_status and 'ol' in ups_status[0].lower():
+                self.middleware.logger.debug(
+                    f'Shutdown not initiated as ups.status ({ups_status[0]}) indicates '
+                    f'{config["identifier"]} is ONLINE (OL).'
+                )
+            else:
+                syslog.syslog(syslog.LOG_INFO, 'upssched-cmd "issuing shutdown"')
```

Comment:
@william-gr we do use syslog in middlewared, maybe we can start getting rid of it now?

Reply:
We do not, we only use file logging.

If I am not mistaken that syslog was supposed to log to console. Weird that I am not seeing an `openlog`